sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/SqlParser.scala

```diff
@@ -58,6 +58,7 @@ class SqlParser extends AbstractSparkSQLParser with DataTypeParser {
   protected val BY = Keyword("BY")
   protected val CASE = Keyword("CASE")
   protected val CAST = Keyword("CAST")
+  protected val DAY = Keyword("DAY")
```

Comment:
can you add a unit test to make sure we can still use "day" as a column name? 

if not, I think we need to make sure the identifier can include day/hour etc